pages/routerGuard.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react';
+import { getSession, signin } from 'next-auth/client';
+import { useRouter } from 'next/router';
+import { DateTime } from 'luxon';
+
+export const RouterGuard: React.FC = ({ children }) => {
+  const [isAuthed, setIsAuthed] = useState(false);
+  const { push, pathname, query, isReady } = useRouter();
+  const getSess = async () => {
+    const session = await getSession();
+
+    setIsAuthed(!!session);
+    // If they do not have a valid session and they're not on the login page, navigate them to it
+    if (!session && pathname !== '/login') {
+      push('/login');
+    }
+    // Check if the session has a user
+    if (session?.user) {
+      // If the session is expired, sign them in again to fresh token
+      // TODO eventually will have to accomidate for okta, google signin, etc
```

Comment:
We'll have to have some logic to determine what provider they authed with, and then resign them in accordingly.  Maybe that's provided by ```next-auth``` or maybe we have to store it somewhere. Maybe in the user object? Also heard mention that Google sign-in could be coming to mpdx in the future, but not sure when.